Core/Src/porting/lib/lzma/lzma.c

```diff
@@ -0,0 +1,36 @@
+#include "lzma.h"
+#include "assert.h"
+
+
+static void *SzAlloc(ISzAllocPtr p, size_t size) {
+    void* res = p->Mem;
+    return res;
+}
+
+static void SzFree(ISzAllocPtr p, void *address) {
+}
+
+const uint8_t lzma_prop_data[5] = {0x5d, 0x00, 0x40, 0x00, 0x00};
+
+void lzma_init_allocs(ISzAlloc *allocs, uint8_t *heap){
+    allocs->Alloc = SzAlloc;
+    allocs->Free = SzFree;
+    allocs->Mem = heap;
+}
+
+size_t lzma_inflate(uint8_t *dst, size_t dst_size, const uint8_t *src, size_t src_size){
+    unsigned char lzma_heap[LZMA_BUF_SIZE];
```

Comment:
This allocates 16256 bytes on the stack. However, we have 20kB of stack memory so it's fine as the stack usage is quite low. Just something to keep in mind.